### ISSUE TITLE
Could not use GPU for training

### ISSUE THREAD

**Issue by hieplpvip** [cofe-ai/fast-gector#1]
Hi,

I'm having trouble with training using an RTX 3090. No matter what I tried, it doesn't seem that training is happening on the GPU.

When I start training, these is a log saying it's using GPU: `setup device: cuda:0`, but GPU memory usage does not change at all, and CPU usage is very high on 1 core.

I'm using conda with Python 3.7, torch 0.11.0, which I setup with this command:

```
conda install pytorch==1.11.0 torchvision==0.12.0 torchaudio==0.11.0 cudatoolkit=11.3 -c pytorch
```

Am I missing anything? Please help me if possible. Thank you!

**Comment by Jason3900** [comment]
Hey, 
That's quite weird. Maybe you can set breakpoints around [here](https://github.com/cofe-ai/fast-gector/blob/364d38b7019aa1b9987a22341d5fb7682d49898b/src/trainer.py#L237) to see what's happening.

**Comment by mrqorib** [comment]
Hi @Jason3900 I also experienced the same issue. The code doesn't use the GPU at all. Also, even though I have set the CUDA_VISIBLE_DEVICES to only 1 GPU, but it somehow detects 8 GPUs even though my machine only has 4 GPUs.
This is the log of the training
`[2023-09-05 09:30:28,750] [INFO] [real_accelerator.py:133:get_accelerator] Setting ds_accelerator to cuda (auto detect)
[2023-09-05 09:30:35,526] [INFO] [runner.py:452:main] Using IP address of 172.28.176.28 for node localhost
[2023-09-05 09:30:35,526] [INFO] [runner.py:555:main] cmd = /home/reza/miniconda3/envs/fast-gector/bin/python -u -m deepspeed.launcher.launch --world_info=eyJsb2NhbGhvc3QiOiBbMCwgMSwgMiwgMywgNCwgNSwgNiwgN119 --master_addr=172.28.176.28 --master_port=49828 --enable_each_rank_log=None train.py --deepspeed --deepspeed_config configs/ds_config_zero1_fp16.json --fp16 --num_epochs 10 --max_num_tokens 128 --valid_batch_size 256 --cold_step_count 0 --warmup 0.1 --cold_lr 1e-3 --skip_correct 0 --skip_complex 0 --sub_token_mode average --special_tokens_fix 1 --unk2keep 0 --tp_prob 1 --tn_prob 1 --detect_vocab_path ./data/vocabulary/d_tags.txt --correct_vocab_path ./data/vocabulary/labels.txt --do_eval --train_path data/clang8.par --valid_path data/bea-full-valid.par --save_dir ckpts/ckpt_20230905_09:30:24 --use_cache 0 --log_interval 1 --eval_interval 50 --save_interval 50 --pretrained_transformer_path roberta-base --tensorboard_dir logs/tb/gector_20230905_09:30:24
[2023-09-05 09:30:37,309] [INFO] [real_accelerator.py:133:get_accelerator] Setting ds_accelerator to cuda (auto detect)
[2023-09-05 09:30:38,142] [INFO] [launch.py:145:main] WORLD INFO DICT: {'localhost': [0, 1, 2, 3, 4, 5, 6, 7]}
[2023-09-05 09:30:38,143] [INFO] [launch.py:151:main] nnodes=1, num_local_procs=8, node_rank=0
[2023-09-05 09:30:38,143] [INFO] [launch.py:162:main] global_rank_mapping=defaultdict(<class 'list'>, {'localhost': [0, 1, 2, 3, 4, 5, 6, 7]})
[2023-09-05 09:30:38,143] [INFO] [launch.py:163:main] dist_world_size=8
[2023-09-05 09:30:38,143] [INFO] [launch.py:165:main] Setting CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7
[2023-09-05 09:30:39,740] [INFO] [real_accelerator.py:133:get_accelerator] Setting ds_accelerator to cuda (auto detect)
[2023-09-05 09:30:40,126] [INFO] [real_accelerator.py:133:get_accelerator] Setting ds_accelerator to cuda (auto detect)
[2023-09-05 09:30:40,142] [INFO] [real_accelerator.py:133:get_accelerator] Setting ds_accelerator to cuda (auto detect)
[2023-09-05 09:30:40,144] [INFO] [real_accelerator.py:133:get_accelerator] Setting ds_accelerator to cuda (auto detect)
[2023-09-05 09:30:40,168] [INFO] [real_accelerator.py:133:get_accelerator] Setting ds_accelerator to cuda (auto detect)
[2023-09-05 09:30:40,211] [INFO] [real_accelerator.py:133:get_accelerator] Setting ds_accelerator to cuda (auto detect)
[2023-09-05 09:30:41,108] [WARNING] [comm.py:152:init_deepspeed_backend] NCCL backend in DeepSpeed not yet implemented
[2023-09-05 09:30:41,108] [INFO] [comm.py:616:init_distributed] cdb=None
[2023-09-05 09:30:41,108] [INFO] [comm.py:643:init_distributed] Initializing TorchBackend in DeepSpeed with backend nccl
[2023-09-05 09:30:41,417] [WARNING] [comm.py:152:init_deepspeed_backend] NCCL backend in DeepSpeed not yet implemented
[2023-09-05 09:30:41,418] [INFO] [comm.py:616:init_distributed] cdb=None
[2023-09-05 09:30:41,450] [WARNING] [comm.py:152:init_deepspeed_backend] NCCL backend in DeepSpeed not yet implemented
[2023-09-05 09:30:41,450] [INFO] [comm.py:616:init_distributed] cdb=None
[2023-09-05 09:30:41,488] [WARNING] [comm.py:152:init_deepspeed_backend] NCCL backend in DeepSpeed not yet implemented
[2023-09-05 09:30:41,488] [INFO] [comm.py:616:init_distributed] cdb=None
[2023-09-05 09:30:41,523] [WARNING] [comm.py:152:init_deepspeed_backend] NCCL backend in DeepSpeed not yet implemented
[2023-09-05 09:30:41,523] [INFO] [comm.py:616:init_distributed] cdb=None
[2023-09-05 09:30:41,523] [WARNING] [comm.py:152:init_deepspeed_backend] NCCL backend in DeepSpeed not yet implemented
[2023-09-05 09:30:41,523] [INFO] [comm.py:616:init_distributed] cdb=None
[2023-09-05 09:30:41,605] [INFO] [real_accelerator.py:133:get_accelerator] Setting ds_accelerator to cuda (auto detect)
[2023-09-05 09:30:42,391] [INFO] [real_accelerator.py:133:get_accelerator] Setting ds_accelerator to cuda (auto detect)
[2023-09-05 09:30:42,905] [WARNING] [comm.py:152:init_deepspeed_backend] NCCL backend in DeepSpeed not yet implemented
[2023-09-05 09:30:42,905] [INFO] [comm.py:616:init_distributed] cdb=None
[2023-09-05 09:30:43,747] [WARNING] [comm.py:152:init_deepspeed_backend] NCCL backend in DeepSpeed not yet implemented
[2023-09-05 09:30:43,747] [INFO] [comm.py:616:init_distributed] cdb=None
[2023-09-05 09:30:43,918] [INFO] [logging.py:96:log_dist] [Rank 0] setup device: cuda:0
[2023-09-05 09:30:43,920] [INFO] [logging.py:96:log_dist] [Rank 3] setup device: cuda:3
[2023-09-05 09:30:43,920] [INFO] [logging.py:96:log_dist] [Rank 1] setup device: cuda:1
[2023-09-05 09:30:43,923] [INFO] [logging.py:96:log_dist] [Rank 6] setup device: cuda:6
[2023-09-05 09:30:43,925] [INFO] [logging.py:96:log_dist] [Rank 5] setup device: cuda:5
[2023-09-05 09:30:43,925] [INFO] [logging.py:96:log_dist] [Rank 4] setup device: cuda:4
[2023-09-05 09:30:43,926] [INFO] [logging.py:96:log_dist] [Rank 2] setup device: cuda:2
[2023-09-05 09:30:43,926] [INFO] [logging.py:96:log_dist] [Rank 7] setup device: cuda:7
Downloading model.safetensors: 100%|██████████| 499M/499M [00:05<00:00, 92.2MB/s]
Some weights of the model checkpoint at roberta-base were not used when initializing RobertaModel: ['lm_head.dense.weight', 'lm_head.dense.bias', 'lm_head.layer_norm.weight', 'lm_head.layer_norm.bias', 'lm_head.bias']
- This IS expected if you are initializing RobertaModel from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).`


Also, may I ask what version of Python are you using to run this code? Many versions of the dependencies are not found on Python 3.7, so I want to try replicating the exact environment.
